Example/Tests/BONUtilitiesTestCase.m

```diff
@@ -0,0 +1,254 @@
+}
+
+- (void)testLabelTextBeforeChainable
+{
+    UILabel *label = UILabel.new;
+    label.textAndApplyChainable = @"Hello, world!";
+    label.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertEqualObjects(label.attributedText.string, @"Hello, world!");
+
+    NSParagraphStyle *defaultParagraphStyle = [[NSParagraphStyle alloc] init];
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(label.attributedText, controlAttributes);
+}
+
+- (void)testLabelAttributedTextAfterChainable
+{
+    UILabel *label = UILabel.new;
+    label.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    label.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+
+    XCTAssertEqualObjects(label.attributedText.string, @"Hello, world!");
+
+    NSDictionary *controlAttributes = [self defaultAttributesForClass:[UILabel class] withString:@"Hello, world!"];
+
+    BONAssertAttributedStringHasAttributes(label.attributedText, controlAttributes);
+}
+
+- (void)testLabelAttributedTextBeforeChainable
+{
+    UILabel *label = UILabel.new;
+    label.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+    label.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertNotNil(label.bonChainable);
+    XCTAssertEqualObjects(label.attributedText.string, @"Hello, world!");
+
+    NSParagraphStyle *defaultParagraphStyle = [[NSParagraphStyle alloc] init];
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(label.attributedText, controlAttributes);
+}
+
+#pragma mark - UITextView
+
+- (void)testTextViewTextAfterChainable
+{
+    UITextView *textView = UITextView.new;
+    textView.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textView.textAndApplyChainable = @"Hello, world!";
+
+    XCTAssertEqualObjects(textView.attributedText.string, @"Hello, world!");
+
+    NSParagraphStyle *defaultParagraphStyle = [[NSParagraphStyle alloc] init];
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+- (void)testTextViewTextBeforeChainable
+{
+    UITextView *textView = UITextView.new;
+    textView.textAndApplyChainable = @"Hello, world!";
+    textView.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
```

Comment:
#? Since a `BONTextable` (as it's soon going to be called) contains its own `string`, it's weird to me that it doesn't overwrite the text in the label. What do you think, @Imperiopolis?